lifting-state-up/lifting-state-up.md

```diff
@@ -0,0 +1,221 @@
+| `isPresentData: !props.isPresent`                   | When we click on the attendance button, the value of `isPresentData` should be the negation of `props.isPresent`.                                                                                                                                              |
+| `props.onUpdate(...);`                              | The value of `props.onUpdate` is a function. We invoke this function using parentheses `()`.                                                                                                                                                                   |
+| `updatedStudent`                                    | The function `props.onUpdate` ultimately accepts one argument, an updated student object. We can pass in our new `updatedStudent` object here.                                                                                                                 |
+
+## Attaching `onAttendanceButtonClick` to the Attendance Button
+
+Now that we have an event handler, let's ensure that we're _using_ this event handler!
+
+We need to attach our function to the attendance button's `onClick` attribute:
+
+<!-- prettier-ignore-start -->
+```js
+const Student = (props) => {
+
+    const onAttendanceButtonClick = () => {
+        const updatedStudent = {
+            id: props.id,
+            nameData: props.name,
+            emailData: props.email,
+            isPresentData: !props.isPresent
+        }
+
+        // Invoke the function passed in through the prop named "onUpate"
+        // This function is referenced by the name "updateStudent" in App
+        props.onUpdate(updatedStudent);
+    }
+
+    const nameColor = props.isPresent ? 'green' : 'red';
+
+    return (
+        <div>
+            <ul>
+                <li className={nameColor}>Nickname: {props.name}</li>
+                <li>Email: {props.email}</li>
+            </ul>
+            <button onClick={onAttendanceButtonClick}>Toggle if {props.name} is present</button>
+        </div>
+    )
+}
+```
+<!-- prettier-ignore-end -->
+
+## Verifying
+
+Let's see if our app works still!
+
+Let's verify that user interactions affect the state of the app using React Dev Tools.
+
+We should first observe that the `App` component indeed keeps `studentData` on state.
+
+![](../assets/lifting-state-up_lifting-state-up_app-state.png)
+
+And clicking on the attendance buttons updates the state in `App`!
+
+![](../assets/lifting-state-up_lifting-state-up_app-state-updated.png)
+
+![]()
+
+We did it! 🎉🎉
+
+Let's observe two more things:
+
+`StudentList` indeed has two `props`: a `students` array with the student data, and `onUpdateStudent`, which is a function.
+
+![](../assets/lifting-state-up_lifting-state-up_studentlist-detail.png)
+
+Each `Student` component has five props:
+
+1. `email`
+1. `id`
+1. `isPresent`
+1. `name`
+1. `onUpdate`, whose value is a function originally named `updateStudentData`
```

Comment:
onUpdate got an explanation, so might as well include it for the others?
reference to a function, and mention the f
```suggestion
1. `email`, a string
1. `id`, a number
1. `isPresent`, a boolean
1. `name`, another string
1. `onUpdate`, a reference to a function, again indicated by the curly 𝑓, originally named `updateStudentData`
```